src/pages/edit-deck-page/edit-deck-page.tsx

```diff
@@ -53,14 +53,20 @@ export const EditDeckPage = () => {
     setDeck(deck);
   }
 
-  function handleCreateDeckClick() {
-    // Todo: create deck
-    // Todo: navigate to view-deck page
-    navigateBackToDecks();
+  async function handleCreateDeckClick(deck: Deck) {
+    const addedDeck = await addDeck(deck);
+    navigate(`${paths.deck}/${addedDeck.metaData.id}`);
   }
 
-  function handleDeleteDeckClick() {
-    // Todo: delete deck
+  async function handleDeleteDeckClick() {
+    if (!isNewDeck && deck !== undefined) {
+      // Todo: always throws. It looks like the response is bad but the deck is still deleted
```

Comment:
oh! i'm 99.99% certain this is on me, delete requests don't have a response so this line would definitely throw:

`const data = response.json()` in `use-fetch-wrapper.ts:39`

you can fix this by changing it to:

```ts
const text = await response.text();
const data = text && JSON.parse(text);
```

Reply:
sweet, that fixed it